### PR TITLE
Add support for integer RJSF type

### PR DESCRIPTION
**Pull request by JSReds**
Change-type: patch
Screenshot 2024-09-24 at 15 27 17
Screenshot 2024-09-24 at 15 32 24
Screenshot 2024-09-24 at 15 32 29

### CODE REVIEW

File: src/components/Filters/FiltersDialog.tsx [hunk 1]
```diff
@@ -284,9 +295,9 @@ export const FiltersDialog = ({
 			schema: {
 				type: 'array',
 				minItems: 1,
-				items: formData.map((data) => {
+				items: formData.map((data, index) => {
 					const schemaField: JSONSchema = {
-						$id: 'field',
+						$id: 'filter' + index + 1,
```
Comment (thgreasi) [inline src/components/Filters/FiltersDialog.tsx:300]: How about v
```suggestion
						$id: `filter-item-${index}`,
```

File: src/components/Filters/FiltersDialog.tsx [hunk 2]
```diff
@@ -321,6 +333,7 @@ export const FiltersDialog = ({
 					return rendererSchema;
 				}),
 				additionalItems: {
+					$id: 'filter' + 0,
```
Comment (thgreasi) [inline src/components/Filters/FiltersDialog.tsx:336]: How about
```suggestion
					$id: 'filter-dialog-additional-item-0',
```

File: src/DataTypes/number.tsx [hunk 3]
```diff
@@ -86,10 +87,14 @@ export const createFilter: CreateFilter<OperatorSlug> = (
...
 
-export const rendererSchema = (schemaField: JSONSchema) => {
+export const rendererSchema = (
+	schemaField: JSONSchema,
+	index: number,
+	propertySchema: JSONSchema,
+) => {
 	const valueSchema: JSONSchema = {
-		type: 'number',
+		type: propertySchema.type,
```
Comment (thgreasi) [inline src/DataTypes/number.tsx:96]: Should we v ?
```suggestion
		type: propertySchema.type ?? 'number',
```

File: src/components/Filters/FiltersDialog.tsx [hunk 4]
```diff
@@ -351,13 +364,17 @@ export const FiltersDialog = ({
 					onSubmit={handleSubmit}
 					{...internalSchemaAndUiSchema}
 					formData={formData}
-					noValidate
 					widgets={widgets}
+					validator={validator}
+					liveValidate={!isFirstValidation}
```
Comment (thgreasi) [inline src/components/Filters/FiltersDialog.tsx:369]: :heart:

File: src/DataTypes/utils.ts [hunk 5]
```diff
@@ -42,12 +43,16 @@ export const getDataTypeSchema = (
 			const: operatorKey,
 		}),
 	);
+	// Let's keep all ids generation here to have a better view.
+	schemaField.$id = `field-${index}`;
+	valueSchema.$id = `value-${index}`;
 	return {
+		$id: (schemaField?.title ?? '') + index,
```
Comment (thgreasi) [inline src/DataTypes/utils.ts:50]: Should we add something as a fallback text for the id here?

Comment (JSReds) [reply src/DataTypes/utils.ts:50]: sure 🤦 schema title is not even appropriate

**Review by thgreasi — APPROVED** [review]

**Pull request merged**